pkg/util/trivy/cache.go

```diff
@@ -46,83 +41,27 @@ func defaultCacheDir() string {
 	return filepath.Join(tmpDir, "trivy")
 }
 
-// NewCustomBoltCache is a CacheProvider. It returns a custom implementation of a BoltDB cache using an LRU algorithm with a
-// maximum number of cache entries, maximum disk size and garbage collection of unused images with its custom cleaner.
-func NewCustomBoltCache(wmeta optional.Option[workloadmeta.Component], cacheDir string, maxDiskSize int) (cache.Cache, CacheCleaner, error) {
+// NewCustomBoltCache returns a BoltDB cache using an LRU algorithm with a
+// maximum disk size and garbage collection of unused images with its custom cleaner.
+func NewCustomBoltCache(wmeta optional.Option[workloadmeta.Component], cacheDir string, maxDiskSize int) (*ScannerCache, *ScannerCacheCleaner, error) {
```

Comment:
Should we return interfaces for `ScannerCache` and `ScannerCacheCleaner` instead of pointer to structure

Reply:
I removed these interfaces for simplicity